From a2bd4332455947037d5451e8ab9be92d7fdfc341 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Mon, 23 Sep 2024 13:29:33 +0300
Subject: [PATCH] [FSW-7256] fs1: define initiator context

Initiator master interface RR page base address
is required to access all RR config registers,
and axuser base is required to access lb_sai
register. These two register areas required
while configuring RR set for any address space
which requires read/write access.
This patch adds support to declare and define
a context with required base addresses.

Change-Id: I0d2356027778565b21faf90f6894c4014e8e174c
---

diff --git a/arch/fs1/arc/dev_security.c b/arch/fs1/arc/dev_security.c
index 7fd0c8c..0df1a37 100644
--- a/arch/fs1/arc/dev_security.c
+++ b/arch/fs1/arc/dev_security.c
@@ -73,6 +73,17 @@
 	}
 };
 
+struct security_initiator_ctx initiator_data[SECURITY_INITIATOR_MAX] = {
+	[SECURITY_INITIATOR_PCIE] = {
+		mmIO_0_CFG_HALF_0_CXL_EP_PCIE_MSTR_IF_LBW_RR_BASE,
+		mmIO_0_CFG_HALF_0_CXL_EP_PCIE_MSTR_IF_LBW_AXI_OVRD_AXUSER_BASE,
+	},
+	[SECURITY_INITIATOR_I2C_S_MSTR] = {
+		mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_I2C_S_MSTR_IF_LBW_RR_BASE,
+		mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_I2C_S_MSTR_IF_LBW_AXI_OVRD_AXUSER_BASE,
+	},
+};
+
 void security_load_revoke_bits(void)
 {
 	uint32_t key_revoke = boot_rom_global.config_fuse & EFUSE_SYSCFG_KEY_REVOCATION_BITS;
diff --git a/arch/fs1/arc/dev_security.h b/arch/fs1/arc/dev_security.h
index 700f579..d52b5b9 100644
--- a/arch/fs1/arc/dev_security.h
+++ b/arch/fs1/arc/dev_security.h
@@ -71,6 +71,11 @@
 	SECURITY_INITIATOR_MAX,
 };
 
+struct security_initiator_ctx {
+	uint32_t base_addr;
+	uint32_t sai_base;
+};
+
 void security_load_revoke_bits(void);
 int security_efuse_image_validation_en(void);
 void boot_rom_eom_overrides(void);
